### DISCUSSION TITLE
How is this different than premake? What does it do better?

### DISCUSSION THREAD

**Discussion by Techcable** [xmake-io/xmake#1863] [Q&A] [answered]
Hi! I'm super interested in this for my C projects. It seems *very* similar to [premake](https://premake.github.io/) just possibly with a slightly wider scope?

What are the advantages compared to say premake (or some other modern build-system like meson)?

The main alternatives seem to be:
- [Meson](https://mesonbuild.com/)
- [Premake](https://premake.github.io/) - Seems *very* similar, also lua-based.

Thanks for your time! :smile:

**Comment by waruqi** [comment, marked as answer]
premake is only a project generator, but xmake is Build backend + Project Generator + Package Manager.

The project generator is just an optional plug-in feature in xmake, it also has many more powerful features, such as package management.

xmake == make/ninja + cmake/premake/meson + vcpkg/conan/conda....

And it has no any dependences, it is very lightweight.

**Reply by Techcable** [reply]
Alright. I think that makes sense :smile:
